doc/build-osx.md

```diff
@@ -145,6 +145,8 @@ It is required that you have `python` and `zip` installed.
 
 ### 1. Configuration
 
+Note: On macOS (x86_64), users with CMake ≥ 4.0 and Homebrew installed in `/usr/local` should pass `-DCMAKE_OSX_SYSROOT=macosx` or `export SDKROOT=macosx` when not building with Homebrew-provided tools.
```

Comment:
This says it applies to users with Homebrew installed, when *not* building with homebrew-provided tools? What are the "tools" here? This seems vauge, and as a user reading this I probably wouldn't know if it applies to me.

Reply:
> This says it applies to users with Homebrew installed, when _not_ building with homebrew-provided tools? What are the "tools" here?

For example, LLVM toolchain.